### PR TITLE
Only show "Loading new file" splash screen when really doing that

### PR DESCRIPTION
**Pull request by cbjeukendrup**
Resolves: #18129 

- When the user just manually opens a new instance*, just show the normal splash screen
- Only when a new instance is opened because the user clicked "new file" inside an existing instance, then we show the "Loading new file" splash screen

\* On Windows, the user can trivially launch new instances manually, by just clicking on the app icon another time. 
On macOS, this is also possible, by typing `open -n /path/to/MuseScore 4.app` in Terminal.

### CODE REVIEW

**Comment by XiaoMigros** [comment]
What happens if the user has the 'Continue last session' preference active? Sorry, I haven't been able to test any newer builds yet.

**Comment by cbjeukendrup** [comment]
Then too, the normal splash screen is shown if the user started the session manually.

**Comment by cbjeukendrup** [comment]
On @bkunda's request, trying out how it would be if there is no splash screen at all when launching a second instance manually.

**Comment by bkunda** [comment]
Tested this on Windows. IMHO this is much better behaviour now: when I open a second instance of the app on Windows, another app window simply appears. There's no additional screens to cause confusion. 

@cbjeukendrup am I right that the changes in this PR won't have any affect on other operating systems (excepting the case of loading a second instance on macOS via the Terminal)? In other words, does this also need to be tested on macOS?

**Comment by cbjeukendrup** [comment]
It is right that there _should_ not be any changes on macOS, but to be on the safe side, it would be good to verify that.

**Comment by bkunda** [comment]
Ok verified that this has not changed anything obvious on macOS (good 👍🏻), so I'm happy with this one @cbjeukendrup!

File: src/appshell/view/internal/splashscreen/splashscreen.h [hunk 1]
```diff
@@ -33,7 +33,7 @@ class SplashScreen : public QWidget
 public:
     enum SplashScreenType {
         Default,
-        ForNewInstance
+        ForSecondaryInstance
```
Comment (Eism) [inline src/appshell/view/internal/splashscreen/splashscreen.h:36]: why did you rename it?
for the third and next will be secondary?

Comment (cbjeukendrup) [reply src/appshell/view/internal/splashscreen/splashscreen.h:36]: I renamed it to make clear that this is only for non-main instances, not launched by the user but by some other instance. In this case, `secondary` means not "second" but "non-main".

File: src/app/app.cpp [hunk 2]
```diff
@@ -168,11 +168,14 @@ int App::run(int argc, char** argv)
     if (runMode == framework::IApplication::RunMode::GuiApp) {
         if (multiInstancesProvider()->isMainInstance()) {
             splashScreen = new SplashScreen(SplashScreen::Default);
-        } else {
-            QString fileName = startupScenario()->startupScoreFile().displayName(true /* includingExtension */);
-            splashScreen = new SplashScreen(SplashScreen::ForNewInstance, fileName);
+        } else if (project::ProjectFile file = startupScenario()->startupScoreFile(); file.isValid()) {
```
Comment (RomanPudashkin) [inline src/app/app.cpp:171]: Please don't use conditions inside if in this way

File: src/appshell/view/internal/splashscreen/splashscreen.h (hunk 1 above)
Comment (Eism) [reply src/appshell/view/internal/splashscreen/splashscreen.h:36]: I would recommend returning it the way it was.
We also use NewInstance in NewInstanceLoadingScreenView

**Review by RomanPudashkin — APPROVED** [review]
